### PR TITLE
[enhance](cloud) Validate and normalize hdfs conf when adding new vault

### PR DESCRIPTION
**Pull request by platoneko**
## Proposed changes

Validate and normalize hdfs conf when adding new vault

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by platoneko** [comment]
run buildall

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 1]
```diff
@@ -308,6 +308,61 @@ static std::string next_available_vault_id(const InstanceInfoPB& instance) {
...
+// Validate and normalize hdfs fs_name. Return true if fs_name is valid.
+bool normalize_hdfs_fs_name(std::string& fs_name) {
+    if (fs_name.empty()) {
+        return false;
+    }
+
+    // Should check scheme existence?
+
+    strip_trailing(fs_name, ' ');
+    strip_leading(fs_name, ' ');
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_resource.cpp:359]: should we consider all `white space` chars?

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 2]
```diff
@@ -308,6 +308,61 @@ static std::string next_available_vault_id(const InstanceInfoPB& instance) {
...
+    size_t start = str.find_first_not_of(c);
+    if (start == std::string::npos) {
+        str = "";
+    } else if (start > 0) {
+        str = str.substr(start);
+    }
+}
+
+// Validate and normalize hdfs prefix. Return true if prefix is valid.
+bool normalize_hdfs_prefix(std::string& prefix) {
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_resource.cpp:334]: Add comment that, if the input is invalid, it is not touched after calling this function.

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 3]
```diff
@@ -308,6 +308,61 @@ static std::string next_available_vault_id(const InstanceInfoPB& instance) {
...
+    }
+
+    strip_trailing(prefix, ' ');
+    strip_leading(prefix, ' ');
+    strip_trailing(prefix, '/');
+    return true;
+}
+
+// Validate and normalize hdfs fs_name. Return true if fs_name is valid.
+bool normalize_hdfs_fs_name(std::string& fs_name) {
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_resource.cpp:351]: ditto

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**